h2o-py/tests/testdir_jira/pyunit_pubdev_5336.py

```diff
@@ -0,0 +1,33 @@
+import h2o
+import pandas as pd
+from tests import pyunit_utils
+
+
+def pubdev_5336():
+    data = pd.DataFrame({'Origin': ['SFO', 'SAN', 'SFO', 'NYC'],
+                         'Dest': ['SFO', 'SFO', 'SAN', 'SAN']})
+    frame = h2o.H2OFrame(data)
+    frame['Origin'].asfactor()
+    frame['Dest'].asfactor()
+
+    # First column has one more categorical variable
+    assert frame['Origin'].nlevels() == [3]
```

Comment:
It would be nicer to check the exact domain, including the order. Eg. if H2O (erroneously) decides to order the domains [SFO, SAN, NYC] and [SFO, SAN] this PR would pass even without this change.

Reply:
Added such tests.